glean-core/ios/GleanTests/DispatchersTest.swift

```diff
@@ -106,4 +106,76 @@ class DispatchersTest: XCTestCase {
             )
         }
     }
+
+    func testCancelBackgroundTasks() {
+        // Set testing mode to false to allow for background execution
+        Dispatchers.shared.setTestingMode(enabled: false)
+
+        // Set task queueing to true to ensure that we clear out the queued tasks when we cancel
+        Dispatchers.shared.setTaskQueuing(enabled: true)
+
+        // Add a task that should be queued
+        Dispatchers.shared.launch {
+            print("A queued task")
+        }
+
+        // Assert the task was queued
+        XCTAssertEqual(Dispatchers.shared.preInitOperations.count, 1, "Task must be queued")
+
+        // Now cancel the tasks
+        Dispatchers.shared.cancelBackgroundTasks()
+
+        // Assert the task was removed from the queue
+        XCTAssertEqual(Dispatchers.shared.preInitOperations.count, 0, "Task must be removed")
+
+        // Set up our test conditions for normal execution by setting queuing to false
+        Dispatchers.shared.setTaskQueuing(enabled: false)
+
+        // Create some counters to use to determine if the tasks were actually cancelled
+        var syncCounter = 0
+        var asyncCounter = 0
+
+        // Create a sync and an async task to add to the Dispatchers that handle the
+        // `isCancelled` property so that they can be cancelled
+        let serialOperation = BlockOperation()
+        serialOperation.addExecutionBlock {
+            while !serialOperation.isCancelled {
+                syncCounter += 1
+            }
+        }
+        Dispatchers.shared.serialOperationQueue.addOperation(serialOperation)
+
+        let concurrentOperation = BlockOperation()
+        concurrentOperation.addExecutionBlock {
+            while !concurrentOperation.isCancelled {
+                asyncCounter += 1
+            }
+        }
+        Dispatchers.shared.launchAsync(operation: concurrentOperation)
+
+        // Let the tasks run for 1 second
```

Comment:
Why is this needed? Can you document the rationale?

Reply:
It's needed just so that the background tasks have some time to run.  I can us uSleep for a shorter delay or attempt to find some other way to guarantee that the tasks are running.

Reply:
No no, that's fine, just extend the comment to mention that.